pydantic_factories/value_generators/regex.py

```diff
@@ -0,0 +1,135 @@
+    "urlsafe": ascii_letters + digits + "-._~",
+    "domainsafe": ascii_letters + digits + "-",
+}
+
+_categories = {
+    "category_digit": _alphabets["digits"],
+    "category_not_digit": _alphabets["nondigits"],
+    "category_space": _alphabets["whitespace"],
+    "category_not_space": _alphabets["nonwhitespace"],
+    "category_word": _alphabets["word"],
+    "category_not_word": _alphabets["nonword"],
+}
+
+
+class RegexFactory:
+    def __init__(self, limit: int = 10, seed: Optional[int] = None) -> None:
+        self._limit = limit
+        self._cache: Dict[str, Any] = {}
+        self._random = Random(x=seed)
+
+        self._cases = {
+            "literal": chr,
+            "not_literal": lambda x: self._random.choice(printable.replace(chr(x), "")),
+            "at": lambda x: "",
+            "in": self._handle_in,
+            "any": lambda x: self._random.choice(printable.replace("\n", "")),
+            "range": lambda x: [chr(i) for i in range(x[0], x[1] + 1)],
+            "category": lambda x: _categories[str(x).lower()],
+            "branch": lambda x: "".join(self._handle_state(i) for i in self._random.choice(x[1])),
+            "subpattern": self._handle_group,
+            "assert": lambda x: "".join(self._handle_state(i) for i in x[1]),
+            "assert_not": lambda x: "",
+            "groupref": lambda x: self._cache[x],
+            "min_repeat": lambda x: self._handle_repeat(*x),
+            "max_repeat": lambda x: self._handle_repeat(*x),
+            "negate": lambda x: [False],
+        }
+
+    def __call__(self, string_or_regex: Union[str, Pattern]) -> str:
+        pattern = string_or_regex.pattern if isinstance(string_or_regex, Pattern) else string_or_regex
+        parsed = parse(pattern)
+        result = self._build_string(parsed)
+        self._cache.clear()
+        return result  # noqa: R504
+
+    def _build_string(self, parsed: SubPattern) -> str:
+        return "".join([self._handle_state(state) for state in parsed])  # type: ignore
+
+    def _handle_state(self, state: Tuple[SubPattern, Tuple[Any, ...]]) -> Any:
+        opcode, value = state
+        return self._cases[str(opcode).lower()](value)  # type: ignore
+
+    def _handle_group(self, value: Tuple[Any, ...]) -> str:
+        result = "".join(self._handle_state(i) for i in value[3])
+        if value[0]:
+            self._cache[value[0]] = result
+        return result
+
+    def _handle_in(self, value: Tuple[Any, ...]) -> Any:
+        candidates = list(chain(*(self._handle_state(i) for i in value)))
+        if candidates and candidates[0] is False:
+            candidates = list(set(printable).difference(candidates[1:]))
+            return self._random.choice(candidates)
+        return self._random.choice(candidates)
+
+    def _handle_repeat(self, start_range: int, end_range: Any, value: SubPattern) -> str:
+        result: List[str] = []
+        end_range = min((end_range, self._limit))
```

Comment:
```suggestion
        end_range = min(end_range, self._limit)
```

Reply:
It is just a consistency suggestions. Two lines below max is used without tuple `max(start_range, end_range)`